### DISCUSSION TITLE
Call to undefined method [waitForLivewireToLoad].

### DISCUSSION THREAD

**Discussion by bhulsman** [livewire/livewire#4099] [1. Help] [answered]
Using Livewire 2.7.x and Dusk 6.19.x I'm getting the following error: `BadMethodCallException: Call to undefined method [waitForLivewireToLoad].`

```
Livewire::visit($browser, LatestNews::class)
    ->assertSee($post1->title)
    ->click('@next-post')
    ->assertSee($post2->title);
```

Related to https://github.com/livewire/livewire/blob/e39edcae6b1971b2d0f327a8e25c40e3d68cb7a0/src/LivewireManager.php#L119

I cannot find anything in the docs to suggest I should manually add this. What am I missing and why does it not know of the method `waitForLivewireToLoad`?

**Comment by joshhanley** [comment, marked as answer]
@bhulsman are you using dusk in a package or in an app?

If using dusk to test package, I extracted everything to test Livewire in a package, into it's own package https://github.com/joshhanley/livewire-dusk-testbench

If using dusk in an app, you will need to configure Livewire's dusk test macros, in your dusk `TestCase` file, add the following lines

```php
use Laravel\Dusk\Browser;
use Livewire\Macros\DuskBrowserMacros;

public function setUp(): void {
    Browser::mixin(new DuskBrowserMacros);
    
    parent::setUp();
    
    // other setup stuff if you have it
}
```

I'm not sure if that will work on it's own though as the `Livewire::visit` is an internal method for using dusk to test Livewire itself.
That's why you won't see any documentation on it.
I think there may be some other stuff is needed to get it running, but test it out and let me know.

Hope this helps!

**Reply by bhulsman** [reply]
Thanks! This seems to fix that issue, only to get another; a timeout. Probably not related to this however.

I would actually prefer to not test components like this, but simply interacting with the webpage and clicking Livewire buttons for instance doesn't work. It would show correctly in the screenshots Dusk makes, but still not pass the test.

Apparently not many people are using Livewire and Dusk, since there is not a lot I can find about it online.

**Reply by joshhanley** [reply]
@bhulsman ok great! Ah bugger.

Yeah, I wonder why that wouldn't work. Maybe it's missing some required config when using the direct route.

Yeah not a lot of people are, but we use it extensively to test Livewire internally.
